### DISCUSSION TITLE
Group automations/scripts/scenes

### DISCUSSION THREAD

**Discussion by mcinnes01** [home-assistant/frontend#8874] [Automations / scripts / scenes] [answered]
**Is your feature request related to a problem? Please describe.**
When you have a lot of automations the UI looks cluttered and its difficult to find the automation you are interested in.

**Describe the solution you'd like**
Add ways to group automations, a field to group the automations by would be the simplest option. This should be used for structuring the display of automation, but it should also enable a search facility to be available. As there is already more information that can be used from the detail in an automation, this should also be surfaced to the search facility such as device/service/group/etc.

Another dimension to this is that from a device/entity/integration etc, you could also provide the ability to display the automations related to that device/entity/integration.

**Describe alternatives you've considered**
I don't think there is a good alternative, the UI doesn't make it easy to manage, view or find automations when you have a fair few of them.

**Comment by zsarnett** [comment]
Had an idea for this before but hadn't gotten around to doing a mockup yet, will share what I had in mind. Basically I think the system can and _should_ handle everything here, where things would just work without any manual organization required on the users part. 

I think having the following three filters would handle like 95% of use cases:
1. **Domain** - filter by any domain. So you can see all automations with a light entity, anything with a lock, person, media player, vacuum, alarm panel, etc.
2. **Device Class** - any binary sensors with motion, door, window, occupancy, etc. So if you wanted to see all automations involving your windows or doors, can do that very easily.
3. **Integration** - with this filter you could see any automation that involves entities belonging to Plex, or Google Cast, or Hue, etc. 

This would make finding things super easy even if you had hundreds of automations. You'd be able to filter the list within just two clicks the majority of the time. These filters would also be tailored to the users personal setup too - if they didn't have any smart locks, lock wouldn't show up under the domain filter for example. So that would keep things clean and manageable. 

The challenging part would be figuring out a good filtering UI. I was thinking when the filter button is pressed, it would show these three filters as "chips" each on their own line, and you'd just horizontally scroll them. But that might not be ideal for users with very large setups that have dozens of integrations, domains, etc.


From SeanPM5 


Add Area From Bram

**Comment by alex6480** [comment]
The filtering sounds very useful. I don't think horizontal scrolling is the way to go as it's often unintuitive to use. It's probably better to just let each filter occupy multiple lines and let the chips wrap so all filters are visible when you press the filter button. Another option is to implement it as a searchable dropdown. If you know what you are looking for you just begin typing the name of the integration and it autocompletes it for you and suggets things along the way. And if you need to, you can use the drop down to scroll through all available values.

I also still think groups could be useful for many people. I find that even simple tasks in Home Assistant often require several automations, so it would be nice to be able to display them together.

I've made a mockup as to how it might work.

Automation page:
Automations without a group could just be shown on the automation page below the groups. That way you don't have to use the group feature if you don't want to.
![automation_panel](https://user-images.githubusercontent.com/8026279/94994302-0a8ac800-0597-11eb-8037-13629a8dedf9.png)

How it looks when you enter a group:
![automation_group](https://user-images.githubusercontent.com/8026279/94994301-09f23180-0597-11eb-9edd-5867c0380e77.png)

How it looks when you are editing an automation:
![edit_automation](https://user-images.githubusercontent.com/8026279/94994300-09599b00-0597-11eb-8bd3-9e24fd41502d.png)

**Reply by kevin-kraus** [reply]
> The filtering sounds very useful. I don't think horizontal scrolling is the way to go as it's often unintuitive to use. It's probably better to just let each filter occupy multiple lines and let the chips wrap so all filters are visible when you press the filter button. Another option is to implement it as a searchable dropdown. If you know what you are looking for you just begin typing the name of the integration and it autocompletes it for you and suggets things along the way. And if you need to, you can use the drop down to scroll through all available values.
> 
> I also still think groups could be useful for many people. I find that even simple tasks in Home Assistant often require several automations, so it would be nice to be able to display them together.
> 
> I've made a mockup as to how it might work.
> 
> Automation page: Automations without a group could just be shown on the automation page below the groups. That way you don't have to use the group feature if you don't want to. ![automation_panel](https://user-images.githubusercontent.com/8026279/94994302-0a8ac800-0597-11eb-8037-13629a8dedf9.png)
> 
> How it looks when you enter a group: ![automation_group](https://user-images.githubusercontent.com/8026279/94994301-09f23180-0597-11eb-9edd-5867c0380e77.png)
> 
> How it looks when you are editing an automation: ![edit_automation](https://user-images.githubusercontent.com/8026279/94994300-09599b00-0597-11eb-8bd3-9e24fd41502d.png)

I would really like to implement this, but I'm not 100% sure if I can do it. 

Is there still any need for this, or does the majority see this issue as fixed due to the new filtering methods?

**Reply by Xitee1** [reply]
I would love to have such a feature. Filtering isn't a proper solution to me.

**Comment by mrand** [comment]
@alex6480 Tags (rather than folders) would allow all options: things can be filtered visually or viewed globally (which might help see things when re-using)

**Comment by LucaDev** [comment]
What if we would add search folders and labels? A search folder could display automations based on user-defined criteria, while labels would provide a good way to actually tag the automation. Seems a bit too overcomplicated for me though.

I would prefer simple folders as @alex6480 described. As long as you're able to create folders inside of another folder, there are basically like tags while keeping everything much more clean looking.

**Comment by networkpotato** [comment]
I would love the ability to group automations into categories of my choice like @alex6480's mockup. For example, I could create a category (or a "folder") where I put automations that has to do with my garage.

**Comment by ghost** [comment]
+1 for this. I'm approaching 50 automations and being able to sort them would be perfect

**Comment by networkpotato** [comment]
This is actually the one reason I have moved all my automations to Node-RED. Not because Home Assistant can't do the same things, but because it's much easier to organize them there.

**Comment by lukastillmann** [comment]
I feel like this folders vs tags discussion has been going on a while. However, as it becomes easier and easier to manage automations in the frontend, the issue becomes more and more urgent. I really feel that we need some way to organize automations. Wether we use a folder or a tags logic might not be as important in the end.  

Has anyone tried implementing a version of this? I am not proficient enough in Python, but I would love to help in some way if I can.

**Comment by Misiu** [comment]
> @alex6480 Tags (rather than folders) would allow all options: things can be filtered visually or viewed globally (which might help see things when re-using)

The advantage of tags is the ability to add more than one. For example, you could add <kbd>Living room</kbd> and <kbd>Light</kbd> tag to automation, this way when filtering you will be able to filter all automations for the living room or all automations that interact with lights.
Folders can't do that.

**Comment by malgoe** [comment]
I don't really see why folders would be any better than tags. The ability to use multiple tags would be great.
The lack of organisation is what keeps me from migrating most of my automations from Node Red. The automation system has really matured lately and this is the piece of the puzzle that is missing for me.

**Comment by frenck** [comment, marked as answer]
Home Assistant 2021.4.0 adds filtering support to the automation list, which is more flexible.

<https://www.home-assistant.io/blog/2021/04/07/release-20214/#filtering-automations-scripts-and-scenes>

**Reply by Codel1417** [reply]
While filtering is good for finding a specific automation it would be nice to pre-organize automations

**Reply by jullit31** [reply]
@Codel1417 I agree.
Also tags would be much more powerful, as you could, for example, write an automation to turn on/off all automation with a certain tag.
IMHO while being able to filter the automation list is a great feature, it is no substitute for a tagging system.

**Reply by noxhirsch** [reply]
I have nearly 200 automations* - filtering for areas or devices/entities doesn't help a lot in my case, because most of the time I'm looking for e.g. all motion/presence/door sensor automations to work on them - not all automations of an area or a specific sensor.
With tags I could filter for automations for lights, motion/presence/door sensors, computer, 3d printer, weather, smartphone, media player, media center, night-mode/away-mode/guest-mode, dashboard tablets, Alexa/Google Assistant, .... that would help a lot.
(*I have to merge some of them with `choose`, but it will still be well above 100 automations)

**Reply by marcelrummens** [reply]
@frenck Came across this comment of you on the community:
https://community.home-assistant.io/t/put-automations-scripts-in-folders-groups/219510/5

Did the filtering on area/device/entity replace this idea, or can we still hope to be able to label/tag/group our automations in the future?

**Reply by frenck** [reply]
> our automations in the future?

@marcelrummens Unlike popular belief: I do not have a crystal ball.

**Reply by sorenlouv** [reply]
@frenck Do you consider the [very popular feature request](https://community.home-assistant.io/t/grouping-automations-on-frontend-for-organization/102771) of "Organising automations" as completed with the introduction of filters in 2021.4.0, or are you still looking into improving this?

Asking because as a new-comer to Home Assistant I already have a hard time keeping track of my automations. I'd like to see the automations affecting my thermostats or the automations reacting to a window or door opening/closing. It seems that I can only filter by a specific device, not a category of devices.

**Reply by frenck** [reply]
This is not the place for this, please use the community forum topic.

Thanks 👍 

../Frenck

**Reply by sorenlouv** [reply]
Okay that's fair but if I just reduce my question to:

> Do you consider the [very popular feature request](https://community.home-assistant.io/t/grouping-automations-on-frontend-for-organization/102771) of "Organising automations" as completed with the introduction of filters in 2021.4.0, or are you still looking into improving this?

It's not clear to me whether this has been implemented or is still being worked on.

**Reply by frenck** [reply]
This discussion has been marked answered and closed. If you have additional feature requests, please use the community forums (which you already linked).

Thanks 👍 

../Frenck

**Comment by dannymertens** [comment]
Hi,

Thanks to @frenck, (https://community.home-assistant.io/t/grouping-automations-on-frontend-for-organization/102771/120) I know that we can add automations to an area, Go to Devices & Services → Entities, search for automation. (yes, with the . dot) click on the automation and add it to the required area, after that you can use the filters in the automation list to filter for these.

So works great for now as a workaround, But if this is already more or less implemented is it then not very easy, to add this option in the automation scripts, helper… pages?

With guys like @bramkragten, @zsarnett  in the team, this is probably a piece of cake to add to a new version.
I really like the new updates to automations, everything gets easier, faster more clear for non developers, and really would love to move back from Node-red to native HA, but the lack of proper organization for automations, scripts, helpers… is for now still a showstopper :weary:

Seeing the focus in streamlining things, [Streamlining Experiences - Home Assistant](https://www.home-assistant.io/blog/2022/01/19/streamlining-experiences/) and the high vote rate for this in the community, it’s quite strange to me why this is not yet implemented.
Off course, I’m not a developer, so maybe I’m missing something here, if so my sincere apologies, but for now the only reason I can think of why this is low on the list, is that you guys, girls off course all write your automations in YAML and probably use packages what off course limits the use of this, but for non developer gods like me this is really a very missed option.

I don’t want to complain, nag about this and hope that this message is not received like this?, I really appreciate the great work you're doing and know that you all are very busy to make HA the best solution there is, and I can’t say thank you enough for that, but only hope that you can bring this back on the table and maybe can put this a little higher on the probably already enormous to-do list?

The ultimate goal would be that we can create some kind of package in the UI, where we can see, in one overview, everything that's linked to that package. 

So for instance, school package

One overview page with all the linked automations, scripts, helpers, templates, custom sensors so that we can easily see, edit in one view what's used for this process, something like this, I'm not a designer, developer but a specialist like @matthiasdebaat can probably even make it look good ;-)

<img width="854" alt="image" src="https://user-images.githubusercontent.com/46963011/174436744-38cc835c-663b-4d8b-ba3c-79eb03d710d6.png">

Thanks in advance!

**Comment by at-karan-sharma** [comment]
I want to add to @dannymertens's post. If we look at multiple recent feedback posts from the community on Reddit ([source1](https://www.reddit.com/r/homeassistant/comments/18xps4j/what_do_you_want_for_2024/), [source2](https://www.reddit.com/r/homeassistant/comments/1afsw20/20242_is_bringing_real_drag_drop_support_for/)), it is clear that the automation/script/helper organization problem is far from fixed.

I, too, deeply appreciate the fact that Home Assistant exists, and provides the power and flexibility of home automation back to users. However, there is still huge potential for improvement to the UX. These improvements would bring in "less advanced" users and also reduce time spent by "advanced" users. Personally, as a somewhat capable user, being visually confronted with a scrollable vertical list of ~30 automations on entering the page is already intimidating.

Correct me if I'm wrong, but there is some uncommunicated reason the popular request for user-defined tags/folders is not being considered. If so, could we discuss the limitations here? I'm happy to contribute towards a full solution to this problem.

**Reply by johntdyer** [reply]
Tags ( [Labels](https://github.com/home-assistant/core/pull/69996))  are coming soon to a release near you ...  Not sure if they will extend to automations in the UI or not... but just wanted to let you know that they are WIP

**Reply by at-karan-sharma** [reply]
Thanks for sharing, that's encouraging news!

I see the inspiration is GitHub Labels, which is certainly an improvement. To build on this, my understanding of the root problem is that long scrolling lists are hard to efficiently parse. I hope the UX design also considers UI structures to group automations on first entering the page, like accordions or folders. That way, users can parse fewer items first and then drill down as needed.
